File: client-scalaz/src/main/scala/org/openapitools/client/api/ApplicationConfigResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime


import ApplicationConfigResponse._

case class ApplicationConfigResponse (
  valid: Option[Boolean],
message: Option[String],
version: Option[Double],
serializeNulls: Option[Boolean],
startTimeLog: Option[Long],
errorCode: Option[String],
request: Option[List[NameStringValueResponse]],
configId: Option[Long],
created: Option[Long],
updated: Option[Long],
configVersion: Option[String],
configJson: Option[String],
configJsonURL: Option[String],
assetId: Option[Long],
returning: Option[String])

object ApplicationConfigResponse {
  import DateTimeCodecs._

  implicit val ApplicationConfigResponseCodecJson: CodecJson[ApplicationConfigResponse] = CodecJson.derive[ApplicationConfigResponse]
  implicit val ApplicationConfigResponseDecoder: EntityDecoder[ApplicationConfigResponse] = jsonOf[ApplicationConfigResponse]
  implicit val ApplicationConfigResponseEncoder: EntityEncoder[ApplicationConfigResponse] = jsonEncoderOf[ApplicationConfigResponse]
}

File: client-clojure/src/sirqul_io_t_platform/specs/application_config_response.clj
(ns sirqul-io-t-platform.specs.application-config-response
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.name-string-value-response :refer :all]
            )
  (:import (java.io File)))


(def application-config-response-data
  {
   (ds/opt :valid) boolean?
   (ds/opt :message) string?
   (ds/opt :version) float?
   (ds/opt :serializeNulls) boolean?
   (ds/opt :startTimeLog) int?
   (ds/opt :errorCode) string?
   (ds/opt :request) (s/coll-of name-string-value-response-spec)
   (ds/opt :configId) int?
   (ds/opt :created) int?
   (ds/opt :updated) int?
   (ds/opt :configVersion) string?
   (ds/opt :configJson) string?
   (ds/opt :configJsonURL) string?
   (ds/opt :assetId) int?
   (ds/opt :returning) string?
   })

(def application-config-response-spec
  (ds/spec
    {:name ::application-config-response
     :spec application-config-response-data}))
